packages/workflow/src/workflow.ts

```diff
@@ -911,11 +911,19 @@ export function proxySinks<T extends Sinks>(): T {
                 const activator = assertInWorkflowContext(
                   'Proxied sinks functions may only be used from a Workflow Execution.'
                 );
+                const info = workflowInfo();
                 activator.sinkCalls.push({
                   ifaceName: ifaceName as string,
                   fnName: fnName as string,
                   // Only available from node 17.
                   args: (globalThis as any).structuredClone ? (globalThis as any).structuredClone(args) : args,
+                  // Clone the workflowInfo object so that any further mutations to it does not get reflected in sink
+                  workflowInfo: {
+                    ...info,
+                    // Make sure to clone any sub-property that may get mutated during the lifespan of an activation
+                    searchAttributes: { ...info.searchAttributes },
+                    memo: info.memo ? { ...info.memo } : undefined,
```

Comment:
I wonder if we can ensure statically that any future nested properties also get copied.
You might want to use `structuredClone` if available instead of manually cloning.

Reply:
That's certainly a possibility, though I'd only consider this if we polyfill `structuredClone` on Node 14 and 16. I really want to avoid leaving that kind of bugs.